### PR TITLE
BUG: writer.wait_closed throws error if connection resets

### PR DESCRIPTION
**Pull request by jdoozer**
Two changes:
1. Tweak the log message for when an `asyncio.IncompleteReadError` exception is caught; from testing with both an async python client and our C++ clients, this actually seems to be the "happy path" when the client closes a thrift connection
2. Add an additional try/except block around the code to close the `writer`, because if the connection has already been closed by the client, trying to close it from within the server will raise a `ConnectionResetError` (at least from what I saw in testing)

### CODE REVIEW

File: aiothrift/server.py [hunk 1]
```diff
@@ -34,13 +34,18 @@ async def __call__(self, reader, writer):
...
         finally:
-            writer.close()
-            await writer.wait_closed()
+            try:
+                writer.close()
+                await writer.wait_closed()
+            except ConnectionResetError:
+                logger.debug(f"client {client_addr} reset connection while trying to close")
+            except Exception:
+                logger.exception("unhandled exception while trying to close client connection")
```
Comment (bagelbits) [inline aiothrift/server.py:48]: Is it worth printing the error?

**Review by bagelbits — APPROVED** [review]
One comment but looks good!

File: aiothrift/server.py (hunk 1 above)
Comment (jdoozer) [reply aiothrift/server.py:48]: hmm yeah probably not... wanted to make sure we caught it but that's about it (I do want the connection reset error msg though)

**Comment by jdoozer** [comment]
Okay after reading more about `close()` and `wait_closed()`, I think we don't need to worry about calling `wait_closed()` at all, and so handling the exception that it can raise is a moot point (note that `close()` swallows exceptions if it can't close the stream b/c it's closed already) - I just pushed a new commit

**Pull request merged**